src/librustc_resolve/lib.rs

```diff
@@ -2319,20 +2319,27 @@ impl<'a> Resolver<'a> {
                     } else if i == 0 {
                         (format!("use of undeclared type or module `{}`", ident), None)
                     } else {
-                        let mut msg =
-                            format!("could not find `{}` in `{}`", ident, path[i - 1].ident);
+                        // Do not emit `PathRoot`(`{{root}}`) as it's unclear.
+                        let ident_str;
+                        let ident_name = if path[i - 1].ident.name == kw::PathRoot {
+                            "the crate root"
+                        } else {
+                            ident_str = format!("`{}`", path[i - 1].ident);
+                            &ident_str
```

Comment:
I'd avoid this trick for simplicity and use `if ... { "the crate root".to_string() } else { format!(...) }`, we only get to here in the erroneous case so we can sacrifice some performance.